dbeam-core/src/main/java/com/spotify/dbeam/args/QueryBuilderArgs.java

```diff
@@ -74,14 +74,14 @@
   }
 
   private static Boolean checkTableName(String tableName) {
-    return tableName.matches("^[a-zA-Z_][a-zA-Z0-9_]*$");
+    return tableName.matches("^[a-zA-Z_][a-zA-Z0-9_\\\\.]*$");
   }
 
   public static QueryBuilderArgs create(String tableName) {
     Preconditions.checkArgument(tableName != null,
                                 "TableName cannot be null");
     Preconditions.checkArgument(checkTableName(tableName),
-                                "'table' must follow [a-zA-Z_][a-zA-Z0-9_]*");
+                                "'table' must follow [a-zA-Z_][a-zA-Z0-9_\\\\.]*");
```

Comment:
Looks good to support `.` in the table name. But:

1. I don't think it is valid to have a table name ending in `.`.
2. Table name is used to derive the avro record name. I am not sure it is valid to have `.` there. Look into adding some unit tests to cover this use case.